integration/connector_test.go

```diff
@@ -0,0 +1,488 @@
+		break
+	}
+}
+
+func (suite *ConnectorSuite) TestCommand() {
+	ws, err := suite.newWSConnection()
+	require.NoError(suite.T(), err)
+	defer ws.Close()
+
+	cmd := commandPayload{}
+	namespace := model.NewNamespacedIDFrom(suite.thingCfg.DeviceID)
+	cmd.Topic = fmt.Sprintf("%s/%s/things/live/messages/%s", namespace.Namespace, namespace.Name, commandName)
+	cmd.Path = fmt.Sprintf("/features/%s/inbox/messages/%s", featureID, commandName)
+	cmd.Value = "request"
+
+	correlationID := uuid.New().String()
+	cmd.Headers = map[string]interface{}{"content-type": "text/plain", "correlation-id": correlationID}
+
+	respCh := suite.beginWSWait(ws, func(payload []byte) bool {
+		resp := &commandResponsePayload{}
+		if err := json.Unmarshal(payload, resp); err != nil {
+			suite.T().Error(err)
+			return true
+		}
+
+		path := strings.ReplaceAll(cmd.Path, "inbox", "outbox")
+		assert.Equal(suite.T(), path, resp.Path)
+		assert.Equal(suite.T(), cmd.Topic, resp.Topic)
+		assert.Equal(suite.T(), resp.Status, http.StatusOK)
+		assert.Equal(suite.T(), fmt.Sprintf(commandResponseFormat, cmd.Value), resp.Value)
+
+		return true
+	})
+
+	err = websocket.JSON.Send(ws, cmd)
+	require.NoError(suite.T(), err)
+
+	require.True(suite.T(), <-respCh, "command response should be received")
+}
+
+func (suite *ConnectorSuite) TestEvent() {
+	suite.testModify("e", "testEvent")
+}
+
+func (suite *ConnectorSuite) TestTelemetry() {
+	suite.testModify("t", "testTelemetry")
+}
+
+func (suite *ConnectorSuite) testModify(channel string, newValue string) {
+	ws, err := suite.newWSConnection()
+	require.NoError(suite.T(), err)
+	defer ws.Close()
+
+	const subAck = "START-SEND-EVENTS:ACK"
+	ackCh := suite.beginWSWait(ws, func(payload []byte) bool {
+		ack := strings.TrimSpace(string(payload))
+		return ack == subAck
+	})
+
+	sub := fmt.Sprintf("START-SEND-EVENTS?filter=like(resource:path,'/features/%s/*')", featureID)
+	err = websocket.Message.Send(ws, sub)
+	require.NoError(suite.T(), err)
+
+	ok := <-ackCh
+	require.True(suite.T(), ok, "acknowledgement %v should be received", subAck)
+
+	namespace := model.NewNamespacedIDFrom(suite.thingCfg.DeviceID)
+	cmd := things.NewCommand(namespace).Twin().
+		FeatureProperty(featureID, propertyName).Modify(newValue)
+
+	msg := cmd.Envelope(protocol.WithResponseRequired(false))
+
+	eventCh := suite.beginWSWait(ws, func(payload []byte) bool {
+		props := make(map[string]interface{})
+
+		err := json.Unmarshal(payload, &props)
+		if err == nil {
+			suite.T().Logf("event received: %v", props)
+
+			return props["topic"] == fmt.Sprintf("%s/%s/things/twin/events/modified", namespace.Namespace, namespace.Name) &&
+				props["path"] == fmt.Sprintf("/features/%s/properties/%s", featureID, propertyName) &&
+				props["value"] == newValue
+
+		}
+
+		suite.T().Logf("error while waiting for event: %v", err)
+		return false
+	})
+
+	err = suite.sendDittoEvent(channel, msg)
+
+	require.NoError(suite.T(), err)
```

Comment:
General issue, it's always a better option to use the version with a message.

Reply:
Do you mean MQTT messages?

Reply:
No, I mean that it's more useful to use: `require.NoError(suite.T(), err, "something useful here")` instead of `require.NoError(suite.T(), err)`